lib/rucio/daemons/storage/consistency/actions.py

```diff
@@ -0,0 +1,805 @@
+# limitations under the License.
+#
+# Authors:
+# - Stefan Piperov <stefan.piperov@cern.ch>, 2020-2021
+# - Igor Mandrichenko <ivm@fnal.gov>, 2020-2021
+
+"""
+Storage-Consistency-Actions is a daemon to delete dark files, and re-subscribe the missing ones, identified previously in a Storage-Consistency-Scanner run.
+"""
+
+import traceback
+
+import logging
+import os
+import socket
+import threading
+from copy import deepcopy
+from datetime import datetime, timedelta
+from random import randint
+from re import match
+import sys 
+import glob
+import json
+import time
+import gzip
+import re
+import csv
+
+
+from rucio.core.rse import list_rses, get_rse_id
+from rucio.rse.rsemanager import lfns2pfns, get_rse_info, parse_pfns
+from rucio.common.config import config_get
+
+from rucio.common.types import InternalAccount, InternalScope
+from rucio.core.replica import __exists_replicas, update_replicas_states
+from rucio.core.quarantined_replica import add_quarantined_replicas
+from rucio.core.monitor import record_gauge, record_counter, record_timer, MultiCounter
+
+
+from rucio.common.utils import daemon_sleep
+from rucio.common import exception
+from rucio.common.logging import formatted_logger, setup_logging
+from rucio.common.exception import DatabaseException, UnsupportedOperation, RuleNotFound
+from rucio.core.heartbeat import live, die, sanity_check
+from rucio.db.sqla.util import get_db_time
+
+
+
+##########################################################################
+### NOTE: these are needed by local version of declare_bad_file_replicas()
+### remove after move to core/replica.py
+from rucio.db.sqla.session import transactional_session
+from rucio.db.sqla.constants import (ReplicaState, BadFilesStatus)
+##########################################################################
+
+graceful_stop = threading.Event()
+
+
+##########################################################################
+### NOTE: declare_bad_file_replicas will be used directly from core/replica.py
+### when handling of DID is added there
+@transactional_session
+def declare_bad_file_replicas(dids, rse_id, reason, issuer,
+     status=BadFilesStatus.BAD, scheme='srm', session=None):
+    """
+    Declare a list of bad replicas.
+
+    :param dids: The list of DIDs.
+    :param rse_id: The RSE id.
+    :param reason: The reason of the loss.
+    :param issuer: The issuer account.
+    :param status: Either BAD or SUSPICIOUS.
+    :param scheme: The scheme of the PFNs.
+    :param session: The database session in use.
+    """
+    unknown_replicas = []
+    replicas = []
+    if True:
+        for did in dids:
+            scope = InternalScope(did['scope'], vo=issuer.vo)
+            name = did['name']
+            __exists, scope, name, already_declared, size =\
+               __exists_replicas(rse_id, scope, name, path=None, session=session)
+            if __exists and ((str(status) == str(BadFilesStatus.BAD) and not\
```

Comment:
can you format this a bit differently so it's easier to read? more linebreaks, vertical alignment, etc., thanks

Reply:
this is for the whole module, btw

Reply:
As I responded in other comments, this whole piece is being removed, once the modified declare_bad_file_replicas() from PR5068 is accepted.
https://github.com/rucio/rucio/pull/5068